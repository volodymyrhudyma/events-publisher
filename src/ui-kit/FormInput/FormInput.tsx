import React, { ChangeEvent } from 'react';
import { IFormControlProps } from 'src/types';
import FormControl from 'src/ui-kit/FormControl';
import Input from 'src/ui-kit/Input';

interface IFormInputProps extends IFormControlProps {
	placeholder: string;
	value: string;
	onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

const FormInput = ({ placeholder, value, onChange, ...rest }: IFormInputProps) => (
	<FormControl {...rest}>
		<Input value={value} onChange={onChange} placeholder={placeholder} />
	</FormControl>
);

FormInput.defaultProps = {
	required: false
};

export default FormInput;
