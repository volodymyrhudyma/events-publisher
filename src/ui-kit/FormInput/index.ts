import Component from './FormInput';

export default Component;
