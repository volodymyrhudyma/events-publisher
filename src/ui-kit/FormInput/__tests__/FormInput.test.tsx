import React from 'react';
import { shallow } from 'enzyme';
import FormInput from '../FormInput';

describe('FormInput component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <FormInput
        label="Label"
        placeholder="Placeholder"
        value=""
        onChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
