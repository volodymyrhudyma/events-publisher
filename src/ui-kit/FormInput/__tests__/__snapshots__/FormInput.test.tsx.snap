// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render correctly 1`] = `
<FormControl
  label="Label"
  required={false}
>
  <Input
    onChange={[MockFunction]}
    placeholder="Placeholder"
    type="text"
    value=""
  />
</FormControl>
`;
